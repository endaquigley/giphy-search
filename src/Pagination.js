import React from "react";
import styled from "styled-components";

const Wrapper = styled.div`
  display: flex;
  align-items: baseline;
  justify-content: space-between;
`;

const Pagination = React.memo(({ page, prevPage, nextPage }) => (
  <Wrapper>
    <button onClick={prevPage} disabled={page === 0}>
      Prev Page
    </button>
    <span>Page {page + 1}</span>
    <button onClick={nextPage} disabled={page === 99}>
      Next Page
    </button>
  </Wrapper>
));

export default Pagination;
