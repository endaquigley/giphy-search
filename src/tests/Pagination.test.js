import React from 'react';
import ReactDOM from 'react-dom';
import Pagination from '../Pagination';

it('renders a disabled previous button on page 1', () => {
  const component = document.createElement('div');
  ReactDOM.render(<Pagination page={ 0 } />, component);

  expect(component.querySelectorAll('button')[0].disabled).toBe(true);
  expect(component.querySelectorAll('button')[1].disabled).toBe(false);

  expect(component).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(component);
});

it('renders an enabled previous button on pages greater than 1', () => {
  const component = document.createElement('div');
  ReactDOM.render(<Pagination page={ 1 } />, component);

  expect(component.querySelectorAll('button')[0].disabled).toBe(false);
  expect(component.querySelectorAll('button')[1].disabled).toBe(false);

  expect(component).toMatchSnapshot();
  ReactDOM.unmountComponentAtNode(component);
});