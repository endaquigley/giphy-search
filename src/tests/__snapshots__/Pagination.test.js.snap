// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component renders a disabled previous button on page 1 1`] = `
<div>
  <div
    class="sc-bdVaJa bvvVRs"
  >
    <button
      disabled=""
    >
      Prev Page
    </button>
    <span>
      Page 
      1
    </span>
    <button>
      Next Page
    </button>
  </div>
</div>
`;

exports[`Pagination Component renders an enabled previous button on pages greater than 1 1`] = `
<div>
  <div
    class="sc-bdVaJa bvvVRs"
  >
    <button>
      Prev Page
    </button>
    <span>
      Page 
      2
    </span>
    <button>
      Next Page
    </button>
  </div>
</div>
`;
